lib/Doctrine/DBAL/Schema/MySqlSchemaManager.php

```diff
@@ -284,4 +285,38 @@ protected function _getPortableTableForeignKeysList($tableForeignKeys)
 
         return $result;
     }
+
+    public function listTableDetails($tableName)
+    {
+        $table = parent::listTableDetails($tableName);
+
+        /** @var MySqlPlatform $platform */
+        $platform = $this->_platform;
+        $sql      = $platform->getListTableMetadataSQL($tableName);
+
+        $tableOptions = $this->_conn->fetchAssoc($sql);
+
+        $table->addOption('engine', $tableOptions['ENGINE']);
+        $table->addOption('collation', $tableOptions['TABLE_COLLATION']);
+        if ($tableOptions['AUTO_INCREMENT'] !== '') {
```

Comment:
These conditions require additional integration test coverage. It's not obvious if the absence of auto-increment is represented by an empty string in MySQL. It may change in other versions or implementations (e.g. MariaDB). I'd better not add any conditions but return the options as they are.

If there's a case when conditional filtering makes sense, we can add it later based on the case.